### PR TITLE
[1/1][docs] Add FCM V1 instructions to Google Service Account Key docs

### PR DESCRIPTION
**Pull request by christopherwalter**
# Why
We need to amend these docs to support using a Google Service Account Key for sending Android notifications via FCM V1. These docs are linked in the CLI when users are configuring Google Service Account Keys for submissions and notifications. (See @quinlanj's comment on this PR in eas-cli: https://github.com/expo/eas-cli/pull/2197#issuecomment-1912581771).

# How
1. Update the top-level intro with an explanation of the two use cases for uploading a Google Service Account Key to EAS.
2. Move existing instructions for `eas submit` into a new subsection
3. Add a subsection for instructions for uploading GSAK for FCM V1

# Test Plan
Get feedback on copy, check that preview markdown looks good

### CODE REVIEW

**Comment by christopherwalter** [comment]
@quinlanj @amandeepmittal this is a pretty rough draft but I was hoping to get your feedback on the copy and layout changes. I'll update with the correct image URLs before landing.

File: creating-google-service-account.md [hunk 1]
```diff
@@ -1,6 +1,20 @@
...
+
+1. submitting your Android app to Google Play Store with `eas submit`
+2. sending Android notifications via `api.expo.dev/push/send` using the FCM V1 protocol
+    - note: beginning in June 2024, all Android notifications must be sent using the FCM V1 protocol
+
+You can configure separate Google Service Account Keys for each of these use cases, or upload a single Google Service Account Key and use it for both use cases (provided that the correct authorizations are in place for that Service Account in Google Cloud Console). 
+
+The steps for each use case are detailed below.
+
+ ## Uplading a Google Service Account Key for Play Store Submissions
```
Comment (quinlanj) [inline creating-google-service-account.md:15]:
```suggestion
 ## Uploading a Google Service Account Key for Play Store Submissions
```

File: creating-google-service-account.md [hunk 2]
```diff
@@ -43,3 +57,66 @@ If you’d like to submit your Android app to Google Play Store with `eas submit
...
+3. Click **Generate New Private Key**, then confirm by clicking **Generate Key**
+
+[<img src="./assets/creating-google-service-account/03-create-service-account.png" width="800" />](./assets/creating-google-service-account/03-create-service-account.png)
+
+4. Securely store the JSON file containing the private key
+
+[<img src="./assets/creating-google-service-account/04-manage-keys.png" width="800" />](./assets/creating-google-service-account/04-manage-keys.png)
+[<img src="./assets/creating-google-service-account/05-create-new-key.png" width="800" />](./assets/creating-google-service-account/05-create-new-key.png)
+[<img src="./assets/creating-google-service-account/06-key-type.png" width="800" />](./assets/creating-google-service-account/06-key-type.png)
+[<img src="./assets/creating-google-service-account/07-key-saved.png" width="800" />](./assets/creating-google-service-account/07-key-saved.png)
```
Comment (quinlanj) [inline creating-google-service-account.md:86]: These existing assets don't look correct in relation to the instructions. 

IIUC, if you have an existing service account for FCMV1 or dealing with the Google Play store submissions, you'd go through the GCP service account workflow. The pictures we currently have in `assets/creating-google-service-account/n.png` are for this GCP workflow. 

But for creating a new service key for FCMV1, we'll need to go through the Firebase console (which is basically a shim over GCP), but that flow will automatically create a service account key with the necessary permissions. So I'd expect the pictures to be from the Firebase console and look something like this:

Screenshot 2024-02-07 at 12 36 15 PM

**Review by quinlanj — CHANGES_REQUESTED** [review]

File: creating-google-service-account.md (hunk 2 above)
Comment (christopherwalter) [reply creating-google-service-account.md]: @quinlanj I've updated the assets, though I still need to fix the "Use existing service account key" flow (specifically the step where you grant the `Firebase Cloud Messaging API Admin` role to the existing service account).

**Review by quinlanj — APPROVED** [review]
🚀

File: creating-google-service-account.md [hunk 3]
```diff
@@ -1,6 +1,20 @@
...
+# Using a Google Service Account Key with EAS
 
-If you’d like to submit your Android app to Google Play Store with `eas submit` you need to create a Google Service Account key. To do this, the following steps are required:
+A Google Service Account Key is a credential in the form of a JSON file that you can obtain via Google Cloud for a particular Google Service Account. 
+
+EAS requires a Google Service Account Key to be uploaded and configured for the following use cases:
+
+1. submitting your Android app to Google Play Store with `eas submit`
+2. sending Android notifications via `api.expo.dev/push/send` using the FCM V1 protocol
+    - note: beginning in June 2024, all Android notifications must be sent using the FCM V1 protocol
```
Comment (amandeepmittal) [inline creating-google-service-account.md:9]:
```suggestion
1. Submitting your Android app to Google Play Store with `eas submit`
2. Sending Android notifications via `api.expo.dev/push/send` using the FCM V1 protocol

> **Note:** Beginning in June 2024, all Android notifications must be sent using the FCM V1 protocol.
```

**Review by amandeepmittal — APPROVED** [review]
Lgtm 👍 Thanks for updating this.

Left one minor suggestion about the callout and being consistent when addressing use cases.

**Pull request merged**